firmware/library/L1_Drivers/pwm.hpp

```diff
@@ -1,13 +1,176 @@
 #pragma once
 
 #include <cstdint>
+#include <cstdio>
+#include "config.hpp"
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/pin_configure.hpp"
 
 class PwmInterface
 {
  public:
+    enum SysClk : uint8_t
+    {
+        kActivatePwm0 = (1 << 5),
+        kActivatePwm1 = (1 << 6)
+    };
+    enum PwmConfigure : uint8_t
+    {
+        kTimerMode = 0,
+        kUsePwmCap0 = (0 << 2),
+        kCounterEnable = (1 << 0),
+        kResetMr0 = (1 << 1),
+        kPwmEnable = (1 << 3)
+    };
     virtual void Initialize(uint32_t frequency_hz = 1'000) = 0;
     virtual void SetDutyCycle(float duty_cycle) = 0;
     virtual float GetDutyCycle() = 0;
     virtual void SetFrequency(uint32_t frequency_hz) = 0;
     virtual uint32_t GetFrequency() = 0;
 };
+
+class Pwm : public PwmInterface
+{
+ public:
+    static constexpr uint8_t kPwmFunction = 0b001;
+    static constexpr uint8_t kGpioFunction = 0b000;
+    // The variable that holds the MR values
+    static constexpr volatile uint32_t * match_register_table[7] = 
```

Comment:
This shouldn't be constexpr, nor defined in the class. This is due to the fact that you need to override this for testing. Put the definition of this in the pwm.cpp file.